### PR TITLE
MINOR: add retries to MusicExample micro services tests

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

File: src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java [hunk 1]
```diff
@@ -222,4 +225,35 @@ public static void sendInventory(List<KeyValue<Product, Integer>> inventory,
       e.printStackTrace();
     }
   }
+
+  public static <T> T getWithRetires(final Invocation.Builder builder,
```
Comment (ableegoldman) [inline src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java]: Should this be getWithRetries?

**Comment by mjsax** [comment]
Updated.

**Comment by mjsax** [comment]
Added one more commit to increase session timeout and test timeout. Based on this failure: https://jenkins.confluent.io/job/confluentinc/job/kafka-streams-examples/job/5.0.x/310/

KafkaStreams starts up at 21:12:21. Later:
```
[2019-03-21 21:12:37,202] INFO [executor-Heartbeat] [GroupCoordinator 0]: Member EmailService-f7543c96-a03f-4c54-9ba7-17f16783bedb-StreamThread-1-consumer-d9267258-7829-46b6-8627-f6382fe42bf2 in group EmailService has failed, removing it from the group (kafka.coordinator.group.GroupCoordinator)
```

It takes 5 more seconds to get into running:
```
[2019-03-21 21:12:42,456] INFO [EmailService-f7543c96-a03f-4c54-9ba7-17f16783bedb-StreamThread-1] stream-client [EmailService-f7543c96-a03f-4c54-9ba7-17f16783bedb] State transition from REBALANCING to RUNNING (org.apache.kafka.streams.KafkaStreams)
```

We run into timeout
```
[2019-03-21 21:12:53,460] INFO [main] stream-client [EmailService-f7543c96-a03f-4c54-9ba7-17f16783bedb] State transition from RUNNING to PENDING_SHUTDOWN (org.apache.kafka.streams.KafkaStreams)
```

```
shouldSendEmailWithValidContents(io.confluent.examples.streams.microservices.EmailServiceTest)  Time elapsed: 63.669 sec  <<< FAILURE!
java.lang.AssertionError: Condition not met within timeout 30000. Email was never sent.
	at io.confluent.examples.streams.microservices.EmailServiceTest.shouldSendEmailWithValidContents(EmailServiceTest.java:69)
```

Increasing session-timeout should avoid dropping out of the group. Also increase test timeout.

File: src/test/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleTest.java [hunk 2]
```diff
@@ -271,31 +273,30 @@ private SongPlayCountBean songCountPlayBean(final Song song, final long plays) {
...
-            genreChartRequest.get(new GenericType<List<SongPlayCountBean>>() {
-            });
+        final List<SongPlayCountBean> chart = MicroserviceTestUtils.getWithRetries(
+            genreChartRequest,
+            new GenericType<List<SongPlayCountBean>>() {},
+            0);
+        System.err.println(chart.size());
         return chart.size() == 5;
       } catch (Exception e) {
+        e.printStackTrace();
```
Comment (guozhangwang) [inline src/test/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleTest.java]: +100.

File: src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java [hunk 3]
```diff
@@ -222,4 +225,35 @@ public static void sendInventory(List<KeyValue<Product, Integer>> inventory,
...
+  }
+
+    public static <T> T getWithRetries(final Invocation.Builder builder,
+                                       final Class<T> clazz,
+                                       int numberOfRetries) {
+        while (true) {
+            try {
+                return builder.get(clazz);
+            } catch (ServerErrorException exception) {
+                if ("HTTP 504 Gateway Timeout".equals(exception.getMessage()) && numberOfRetries-- > 0) {
```
Comment (guozhangwang) [inline src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java]: I'm wondering if we can relax the check a bit to, say, `exception.getMessage().contains("504")`? I'm not expert in http get / put APIs but it seems various error messages may be returned: https://www.lifewire.com/504-gateway-timeout-error-explained-2622941

**Comment by mjsax** [comment]
Run into timeout:
```
Caused by: org.apache.kafka.common.errors.SerializationException: Error registering Avro schema: {"type":"record","name":"Song","namespace":"io.confluent.examples.streams.avro","fields":[{"name":"id","type":"long"},{"name":"album","type":{"type":"string","avro.java.string":"String"}},{"name":"artist","type":{"type":"string","avro.java.string":"String"}},{"name":"name","type":{"type":"string","avro.java.string":"String"}},{"name":"genre","type":{"type":"string","avro.java.string":"String"}}]}
Caused by: io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException: Register operation timed out; error code: 50002
```

Was looking into this, and it seems we only increased SR timeout for some test but not for all. Added a fix.

File: src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java (hunk 3 above)
Comment (mjsax) [reply src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java]: Sure.

**Comment by mjsax** [comment]
Failures seem to be environmental.

Retest this please.

**Pull request merged**